From 5f1196d76245e9ba346a7bf2fd3287622cb53d24 Mon Sep 17 00:00:00 2001
From: Lotus Developer <dev@lotus.org>
Date: Thu, 16 Apr 2026 12:00:00 +0000
Subject: [PATCH 03/25] chain: brand daemon and RPC client targets as Lotus

Brand the daemon and RPC client binaries and build targets as Lotus while
retaining upstream version metadata and copyright notices.
---
 CMakeLists.txt                                | 10 ++---
 doc/man/CMakeLists.txt                        |  4 +-
 doc/rpc/CMakeLists.txt                        |  8 ++--
 electrum/CMakeLists.txt                       |  4 +-
 src/CMakeLists.txt                            | 40 +++++++++----------
 src/bitcoin-cli.cpp                           | 18 ++++-----
 src/bitcoind.cpp                              |  8 ++--
 src/clientversion.cpp                         |  6 +--
 test/CMakeLists.txt                           |  4 +-
 .../test_framework/test_framework.py          |  4 +-
 10 files changed, 53 insertions(+), 53 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fb513579e..0788025d39 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,10 +6,10 @@ set(CMAKE_USER_MAKE_RULES_OVERRIDE
 	"${CMAKE_SOURCE_DIR}/cmake/modules/OverrideInitFlags.cmake"
 )
 
-project(bitcoin-abc
+project(lotus
 	VERSION 0.34.1
-	DESCRIPTION "Bitcoin ABC is a full node implementation of the eCash protocol."
-	HOMEPAGE_URL "https://www.bitcoinabc.org"
+	DESCRIPTION "Lotus is a full node implementation of the Lotus protocol."
+	HOMEPAGE_URL "https://github.com/givelotus/lotus"
 )
 
 add_custom_target(print-version
@@ -23,8 +23,8 @@ add_custom_target(print-project-name
 )
 
 # Package information
-set(PACKAGE_NAME "Bitcoin ABC")
-set(PACKAGE_BUGREPORT "https://github.com/Bitcoin-ABC/bitcoin-abc/issues")
+set(PACKAGE_NAME "Lotus")
+set(PACKAGE_BUGREPORT "https://github.com/givelotus/lotus/issues")
 
 # Copyright
 set(COPYRIGHT_YEAR 2026)
diff --git a/doc/man/CMakeLists.txt b/doc/man/CMakeLists.txt
index 4a7c0341df..8cc552c9b0 100644
--- a/doc/man/CMakeLists.txt
+++ b/doc/man/CMakeLists.txt
@@ -18,11 +18,11 @@ function(generate_manpages)
 				"${CMAKE_SOURCE_DIR}/cmake/utils/log-and-print-on-failure.sh"
 				"${CMAKE_BINARY_DIR}/doc/man/gen-manpage-${TARGET}.log"
 				"${CMAKE_SOURCE_DIR}/doc/man/gen-manpages.sh"
-				"$<TARGET_FILE:bitcoind>"
+				"$<TARGET_FILE:lotusd>"
 				"$<TARGET_FILE:${TARGET}>"
 				"${MAN_PAGE}"
 			DEPENDS
-				bitcoind
+				lotusd
 				"${TARGET}"
 		)
 		add_custom_target(doc-manpage-${TARGET}
diff --git a/doc/rpc/CMakeLists.txt b/doc/rpc/CMakeLists.txt
index d2c26d3cad..4a4bcba41d 100644
--- a/doc/rpc/CMakeLists.txt
+++ b/doc/rpc/CMakeLists.txt
@@ -6,12 +6,12 @@ if(BUILD_CLI)
 		COMMAND
 			"${CMAKE_CURRENT_SOURCE_DIR}/gen-rpc-docs.sh"
 			"${CMAKE_CURRENT_SOURCE_DIR}/generate.go"
-			"$<TARGET_FILE:bitcoind>"
-			"$<TARGET_FILE:bitcoin-cli>"
+			"$<TARGET_FILE:lotusd>"
+			"$<TARGET_FILE:lotus-cli>"
 		WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
 		DEPENDS
-			bitcoind
-			bitcoin-cli
+			lotusd
+			lotus-cli
 	)
 endif()
 
diff --git a/electrum/CMakeLists.txt b/electrum/CMakeLists.txt
index 7ae6dbcd7b..6985f3fd6b 100644
--- a/electrum/CMakeLists.txt
+++ b/electrum/CMakeLists.txt
@@ -18,10 +18,10 @@ if(BUILD_CHRONIK)
 	add_custom_target(check-electrum-functional
 		COMMENT "Run Electrum ABC functional tests..."
 		COMMAND
-			"BITCOIND=${CMAKE_BINARY_DIR}/src/bitcoind"
+			"BITCOIND=${CMAKE_BINARY_DIR}/src/lotusd"
 			"${Python_EXECUTABLE}" ./test_runner.py functional
 		DEPENDS
-			bitcoind
+			lotusd
 			${CMAKE_CURRENT_BINARY_DIR}/test_runner.py
 	)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0b14b0c0ea..923f185159 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Copyright (c) 2017 The Bitcoin developers
 
-project(bitcoind)
+project(lotusd)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -16,7 +16,7 @@ endforeach()
 
 option(BUILD_WALLET "Activate the wallet functionality" ON)
 option(BUILD_ZMQ "Activate the ZeroMQ functionalities" ON)
-option(BUILD_CLI "Build bitcoin-cli" ON)
+option(BUILD_CLI "Build lotus-cli" ON)
 option(BUILD_TX "Build bitcoin-tx" ON)
 option(BUILD_QT "Build bitcoin-qt" ON)
 option(BUILD_SEEDER "Build bitcoin-seeder" ON)
@@ -787,21 +787,21 @@ if(BUILD_SEEDER)
 	add_subdirectory(seeder)
 endif()
 
-# bitcoin-cli
+# lotus-cli
 if(BUILD_CLI)
-	add_executable(bitcoin-cli bitcoin-cli.cpp)
-	generate_windows_version_info(bitcoin-cli
+	add_executable(lotus-cli bitcoin-cli.cpp)
+	generate_windows_version_info(lotus-cli
 		DESCRIPTION "JSON-RPC client for ${PACKAGE_NAME}"
 	)
 
-	target_link_libraries(bitcoin-cli common rpcclient)
-	link_event(bitcoin-cli event)
+	target_link_libraries(lotus-cli common rpcclient)
+	link_event(lotus-cli event)
 
-	add_to_symbols_check(bitcoin-cli)
-	add_to_security_check(bitcoin-cli)
+	add_to_symbols_check(lotus-cli)
+	add_to_security_check(lotus-cli)
 
-	install_target(bitcoin-cli)
-	install_manpages(bitcoin-cli)
+	install_target(lotus-cli)
+	install_manpages(lotus-cli)
 endif()
 
 # bitcoin-tx
@@ -991,17 +991,17 @@ if(BUILD_CHAINSTATE)
 	add_to_security_check(bitcoin-chainstate)
 endif()
 
-# bitcoind
-add_executable(bitcoind bitcoind.cpp)
-target_link_libraries(bitcoind server)
-generate_windows_version_info(bitcoind
-	DESCRIPTION "Bitcoin node with a JSON-RPC server"
+# lotusd
+add_executable(lotusd bitcoind.cpp)
+target_link_libraries(lotusd server)
+generate_windows_version_info(lotusd
+	DESCRIPTION "Lotus node with a JSON-RPC server"
 )
-add_to_symbols_check(bitcoind)
-add_to_security_check(bitcoind)
+add_to_symbols_check(lotusd)
+add_to_security_check(lotusd)
 
-install_target(bitcoind)
-install_manpages(bitcoind)
+install_target(lotusd)
+install_manpages(lotusd)
 
 # Bitcoin-qt
 if(BUILD_QT)
diff --git a/src/bitcoin-cli.cpp b/src/bitcoin-cli.cpp
index 91cd191899..53717d7708 100644
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -82,7 +82,7 @@ static void SetupCliArgs(ArgsManager &argsman) {
             "arguments are number of blocks to generate (default: %s) and "
             "maximum iterations to try (default: %s), equivalent to RPC "
             "generatetoaddress nblocks and maxtries arguments. Example: "
-            "bitcoin-cli -generate 4 1000",
+            "lotus-cli -generate 4 1000",
             DEFAULT_NBLOCKS, DEFAULT_MAX_TRIES),
         ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg(
@@ -160,7 +160,7 @@ static void SetupCliArgs(ArgsManager &argsman) {
     argsman.AddArg(
         "-rpcwallet=<walletname>",
         "Send RPC for non-default wallet on RPC server (needs to exactly match "
-        "corresponding -wallet option passed to bitcoind). This changes the "
+        "corresponding -wallet option passed to lotusd). This changes the "
         "RPC endpoint used, e.g. http://127.0.0.1:8332/wallet/<walletname>",
         ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
 }
@@ -211,14 +211,14 @@ static int AppInitRPC(int argc, char *argv[]) {
             strUsage += FormatParagraph(LicenseInfo());
         } else {
             strUsage += "\n"
-                        "Usage:  bitcoin-cli [options] <command> [params]  "
+                        "Usage:  lotus-cli [options] <command> [params]  "
                         "Send command to " PACKAGE_NAME "\n"
-                        "or:     bitcoin-cli [options] -named <command> "
+                        "or:     lotus-cli [options] -named <command> "
                         "[name=value]...  Send command to " PACKAGE_NAME
                         " (with named arguments)\n"
-                        "or:     bitcoin-cli [options] help                "
+                        "or:     lotus-cli [options] help                "
                         "List commands\n"
-                        "or:     bitcoin-cli [options] help <command>      Get "
+                        "or:     lotus-cli [options] help <command>      Get "
                         "help for a command\n";
 
             strUsage += "\n" + gArgs.GetHelpMessage();
@@ -510,7 +510,7 @@ public:
 
         const UniValue &networkinfo{batch[ID_NETWORKINFO]["result"]};
         if (networkinfo["version"].getInt<int>() < 230000) {
-            throw std::runtime_error("-netinfo requires bitcoind server to be "
+            throw std::runtime_error("-netinfo requires lotusd server to be "
                                      "running v0.23.0 and up");
         }
 
@@ -813,7 +813,7 @@ static UniValue CallRPC(BaseRequestHandler *rh, const std::string &strMethod,
         }
         throw CConnectionFailed(
             strprintf("Could not connect to the server %s:%d%s\n\nMake sure "
-                      "the bitcoind server is running and that you are "
+                      "the lotusd server is running and that you are "
                       "connecting to the correct RPC port.",
                       host, port, responseErrorMessage));
     } else if (response.status == HTTP_UNAUTHORIZED) {
@@ -919,7 +919,7 @@ static void ParseError(const UniValue &error, std::string &strPrint,
         if (err_code.isNum() &&
             err_code.getInt<int>() == RPC_WALLET_NOT_SPECIFIED) {
             strPrint += "\nTry adding \"-rpcwallet=<filename>\" option to "
-                        "bitcoin-cli command line.";
+                        "lotus-cli command line.";
         }
     } else {
         strPrint = "error: " + error.write();
diff --git a/src/bitcoind.cpp b/src/bitcoind.cpp
index 16d1201338..8efba0cb73 100644
--- a/src/bitcoind.cpp
+++ b/src/bitcoind.cpp
@@ -162,7 +162,7 @@ static bool AppInit(NodeContext &node, int argc, char *argv[]) {
         if (args.IsArgSet("-version")) {
             strUsage += FormatParagraph(LicenseInfo()) + "\n";
         } else {
-            strUsage += "\nUsage:  bitcoind [options]                     "
+            strUsage += "\nUsage:  lotusd [options]                     "
                         "Start " PACKAGE_NAME "\n";
             strUsage += "\n" + args.GetHelpMessage();
         }
@@ -201,12 +201,12 @@ static bool AppInit(NodeContext &node, int argc, char *argv[]) {
             if (!IsSwitchChar(argv[i][0])) {
                 return InitError(Untranslated(
                     strprintf("Command line contains unexpected token '%s', "
-                              "see bitcoind -h for a list of options.",
+                              "see lotusd -h for a list of options.",
                               argv[i])));
             }
         }
 
-        // -server defaults to true for bitcoind but not for the GUI so do this
+        // -server defaults to true for lotusd but not for the GUI so do this
         // here
         args.SoftSetBoolArg("-server", true);
         // Set this early so that parameter interactions go to console
@@ -311,7 +311,7 @@ int main(int argc, char *argv[]) {
 
     SetupEnvironment();
 
-    // Connect bitcoind signal handlers
+    // Connect lotusd signal handlers
     noui_connect();
 
     return (AppInit(node, argc, argv) ? node.exit_status.load() : EXIT_FAILURE);
diff --git a/src/clientversion.cpp b/src/clientversion.cpp
index 8df98f99d2..b5a7f7159c 100644
--- a/src/clientversion.cpp
+++ b/src/clientversion.cpp
@@ -16,7 +16,7 @@
  * for both bitcoind and bitcoin-qt, to make it harder for attackers to
  * target servers or GUI users specifically.
  */
-const std::string CLIENT_NAME("Bitcoin ABC");
+const std::string CLIENT_NAME("Lotus");
 
 #ifdef HAVE_BUILD_INFO
 #include <obj/build.h>
@@ -92,8 +92,8 @@ std::string CopyrightHolders(const std::string &strPrefix) {
 
 std::string LicenseInfo() {
     const std::string URL_SOURCE_CODE =
-        "<https://github.com/Bitcoin-ABC/bitcoin-abc>";
-    const std::string URL_WEBSITE = "<https://www.bitcoinabc.org>";
+        "<https://github.com/givelotus/lotus>";
+    const std::string URL_WEBSITE = "<https://github.com/givelotus/lotus>";
 
     return CopyrightHolders(strprintf(_("Copyright (C) %i-%i").translated, 2009,
                                       COPYRIGHT_YEAR) +
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index c11a6959c9..57e437863f 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -26,7 +26,7 @@ include(TestSuite)
 
 set(_TEST_TARGET_DEPENDS "")
 if(BUILD_CLI)
-	list(APPEND _TEST_TARGET_DEPENDS bitcoin-cli)
+	list(APPEND _TEST_TARGET_DEPENDS lotus-cli)
 endif()
 if(BUILD_WALLET)
 	list(APPEND _TEST_TARGET_DEPENDS bitcoin-wallet)
@@ -73,7 +73,7 @@ function(add_functional_test_check TARGET DESCRIPTION)
 		CUSTOM_TARGET_ARGS
 			COMMENT "Running ${DESCRIPTION}"
 			DEPENDS
-				bitcoind
+				lotusd
 				${_TEST_TARGET_DEPENDS}
 				${CMAKE_CURRENT_BINARY_DIR}/functional/test_runner.py
 			USES_TERMINAL
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index 7681219dae..427a176761 100644
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -333,12 +333,12 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
         fname_bitcoind = os.path.join(
             config["environment"]["BUILDDIR"],
             "src",
-            f"bitcoind{config['environment']['EXEEXT']}",
+            f"lotusd{config['environment']['EXEEXT']}",
         )
         fname_bitcoincli = os.path.join(
             config["environment"]["BUILDDIR"],
             "src",
-            f"bitcoin-cli{config['environment']['EXEEXT']}",
+            f"lotus-cli{config['environment']['EXEEXT']}",
         )
         self.options.bitcoind = os.getenv("BITCOIND", default=fname_bitcoind)
         self.options.bitcoincli = os.getenv("BITCOINCLI", default=fname_bitcoincli)
-- 
2.50.1 (Apple Git-155)

